src/proposal.mligo

```diff
@@ -277,84 +282,102 @@ let do_total_vote_meet_quorum_threshold (proposal, store: proposal * storage): b
   (nil_op, store)
 
+(*
+ * Unstake proposer token and execute decision lambda based on if the proposal is passed or not.
+ *)
 [@inline]
 let handle_proposal_is_over
-    (config, start_level, proposal_key, store, ops, counter
-      : config * blocks * proposal_key * storage * operation list * counter
+    (config, proposal, store, ops
+      : config * proposal * storage * operation list
     )
-    : (operation list * storage * counter) =
-  let proposal = fetch_proposal (proposal_key, store) in
-
-  if is_proposal_age (proposal, config.proposal_expired_level)
-  then (failwith expired_proposal : (operation list * storage * counter))
-  else if is_proposal_age (proposal, config.proposal_flush_level)
-       && counter.current < counter.total // not finished
-  then
-    let counter = { counter with current = counter.current + 1n } in
-    let cond =    do_total_vote_meet_quorum_threshold(proposal, store)
-              && proposal.upvotes > proposal.downvotes
-    in
-    let store = unstake_proposer_token
-          (config.rejected_proposal_slash_value, cond, proposal, config.period, config.fixed_proposal_fee_in_token, store) in
-    let (new_ops, store) =
-      if cond
-      then
-        let dl_out = config.decision_lambda { proposal = proposal; extras = store.extra } in
-        let guardian = match dl_out.guardian with
-          | Some g -> g
-          | None -> store.guardian
-        in (dl_out.operations,
-              { store with extra = dl_out.extras
-              ; guardian = guardian
-              })
-      else (nil_op, store)
-    in
-    let cons = fun (l, e : operation list * operation) -> e :: l in
-    let ops = List.fold cons ops new_ops in
-    let store = remove_from_proposal_sort_by_level (start_level, proposal_key, store) in
-    (ops, store, counter)
-  else (ops, store, counter)
+    : (operation list * storage) =
+  let cond = do_total_vote_meet_quorum_threshold (proposal, store)
+          && proposal.upvotes > proposal.downvotes
+  in
+  let store = unstake_proposer_token
+        (config.rejected_proposal_slash_value, cond, proposal, config.period, config.fixed_proposal_fee_in_token, store) in
+  let (new_ops, store) =
+    if cond
+    then
+      let dl_out = config.decision_lambda { proposal = proposal; extras = store.extra } in
+      let guardian = match dl_out.guardian with
+        | Some g -> g
+        | None -> store.guardian
+      in (dl_out.operations,
+            { store with extra = dl_out.extras
+            ; guardian = guardian
+            })
+    else (nil_op, store)
+  in
+  let cons = fun (l, e : operation list * operation) -> e :: l in
+  let ops = List.fold cons ops new_ops in
+  (ops, store)
+
+
+(*
+ * Flush a proposal, starting from the first. Remove it when the flush is done, and move to the next proposal.
+ *)
+let rec flush_one (counter, config, store, ops : counter * config * storage * operation list): (operation list * storage * counter) =
```

Comment:
Minor nitpick: I think this should be renamed, given that it no longer flushes one proposal, but one proposal per iteration.

Maybe something like `flush_each` would be better.

Reply:
Renamed this.